File: libs/eave-stdlib-ts/src/core-api/operations/subscriptions.ts
import { RequestArgsTeamId, makeRequest } from "../../requests.js";
import { DocumentReference, DocumentReferenceInput, Subscription, SubscriptionInput } from "../models/subscriptions.js";
import { Team } from "../models/team.js";
import { CoreApiEndpointConfiguration } from "./shared.js";

export type CreateSubscriptionRequestBody = {
  subscription: SubscriptionInput;
  document_reference?: DocumentReferenceInput;
};

export type CreateSubscriptionResponseBody = {
  team: Team;
  subscription: Subscription;
  document_reference?: DocumentReference;
};

export class CreateSubscriptionOperation {
  static config = new CoreApiEndpointConfiguration({ path: "/subscriptions/create" });
  static async perform(args: RequestArgsTeamId & { input: CreateSubscriptionRequestBody }): Promise<CreateSubscriptionResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <CreateSubscriptionResponseBody>await resp.json();
    return responseData;
  }
}

export type DeleteSubscriptionRequestBody = {
  subscription: SubscriptionInput;
};

export class DeleteSubscriptionOperation {
  static config = new CoreApiEndpointConfiguration({ path: "/subscriptions/delete" });
  static async perform(args: RequestArgsTeamId & { input: DeleteSubscriptionRequestBody }): Promise<null> {
    await makeRequest({
      config: this.config,
      ...args,
    });
    return null;
  }
}

export type GetSubscriptionRequestBody = {
  subscription: SubscriptionInput;
};

export type GetSubscriptionResponseBody = {
  team: Team;
  subscription?: Subscription;
  document_reference?: DocumentReference;
};

export class GetSubscriptionOperation {
  static config = new CoreApiEndpointConfiguration({ path: "/subscriptions/query" });
  static async perform(args: RequestArgsTeamId & { input: GetSubscriptionRequestBody }): Promise<GetSubscriptionResponseBody> {
    const resp = await makeRequest({
      config: this.config,
      ...args,
    });
    const responseData = <GetSubscriptionResponseBody>await resp.json();
    return responseData;
  }
}

File: libs/eave-stdlib-py/src/eave/stdlib/core_api/operations/subscriptions.py
from typing import Optional, Unpack
import uuid
from eave.stdlib.core_api.models.subscriptions import DocumentReference
from eave.stdlib.core_api.models.subscriptions import Subscription
from eave.stdlib.core_api.models.subscriptions import DocumentReferenceInput
from eave.stdlib.core_api.models.subscriptions import SubscriptionInput
from . import BaseRequestBody, BaseResponseBody, CoreApiEndpoint, CoreApiEndpointConfiguration

from ..models import team
from ... import requests


# class GetSubscriptionRequest(CoreApiEndpoint):
#     config = CoreApiEndpointConfiguration(
#         path="/subscriptions/query",
#         auth_required=False,
#     )

#     class RequestBody(BaseRequestBody):
#         subscription: SubscriptionInput

#     class ResponseBody(BaseResponseBody):
#         team: team.Team
#         subscription: Optional[Subscription]
#         document_reference: Optional[DocumentReference]

#     @classmethod
#     async def perform(
#         cls,
#         input: RequestBody,
#         team_id: uuid.UUID,
#         **kwargs: Unpack[requests.CommonRequestArgs],
#     ) -> ResponseBody:
#         response = await requests.make_request(
#             config=cls.config,
#             input=input,
#             team_id=team_id,
#             **kwargs,
#         )

#         body = await cls.make_response(response, cls.ResponseBody)
#         return body


# class CreateSubscriptionRequest(CoreApiEndpoint):
#     config = CoreApiEndpointConfiguration(
#         path="/subscriptions/create",
#         auth_required=False,
#     )

#     class RequestBody(BaseRequestBody):
#         subscription: SubscriptionInput
#         document_reference: Optional[DocumentReferenceInput] = None

#     class ResponseBody(BaseResponseBody):
#         team: team.Team
#         subscription: Subscription
#         document_reference: Optional[DocumentReference]

#     @classmethod
#     async def perform(
#         cls,
#         input: RequestBody,
#         team_id: uuid.UUID,
#         **kwargs: Unpack[requests.CommonRequestArgs],
#     ) -> ResponseBody:
#         response = await requests.make_request(
#             config=cls.config,
#             input=input,
#             team_id=team_id,
#             **kwargs,
#         )

#         body = await cls.make_response(response, cls.ResponseBody)
#         return body


# class DeleteSubscriptionRequest(CoreApiEndpoint):
#     config = CoreApiEndpointConfiguration(
#         path="/subscriptions/delete",
#         auth_required=False,
#     )

#     class RequestBody(BaseRequestBody):
#         subscription: SubscriptionInput

#     class ResponseBody(BaseResponseBody):
#         pass

#     @classmethod
#     async def perform(
#         cls,
#         input: RequestBody,
#         team_id: uuid.UUID,
#         **kwargs: Unpack[requests.CommonRequestArgs],
#     ) -> ResponseBody:
#         response = await requests.make_request(
#             config=cls.config,
#             input=input,
#             team_id=team_id,
#             **kwargs,
#         )

#         body = await cls.make_response(response, cls.ResponseBody)
#         return body
